slmsuite/hardware/slms/slm.py

```diff
@@ -202,35 +220,61 @@ def write(
+            depend on :meth:`numpy.mod()`.
+
+         - :attr:`phase_scaling` is less than one.
+
+            In this case, the SLM has **more phase tuning range** than necessary.
+            If the data is within the SLM range ``[0, 2*pi/phase_scaling]``, then the data is passed directly.
+            Otherwise, the data is wrapped by :math:`2\pi` using the very slow :meth:`numpy.mod()`.
+            Try to avoid this in applications where speed is important.
+
+         - :attr:`phase_scaling` is more than one.
+
+            In this case, the SLM has **less phase tuning range** than necessary.
+            Processed the same way as the :attr:`phase_scaling` is less than one case, with the
+            important exception that phases (after wrapping) between ``2*pi/phase_scaling`` and
+            ``2*pi`` are set to zero. For instance, a sawtooth blaze would be truncated at the tips.
 
         Parameters
         ----------
         phase : numpy.ndarray or None
-            Data to display in units of phase delay. Data must be larger than the SLM. If larger,
-            the data is cropped to size in a centered manner. If ``None``, data is zeroed.
-            Usually, this is an exact stored copy of the data passed by the user. However, in cases
-            where :attr:`phase_scaling` not one, this copy is modified to include how the data was wrapped.
+            Data to display in units of phase delay (unless the passed data is of integer type).
+
+             - If ``None`` is passed to :meth:`.write()`, data is zeroed.
+             - If the array has a larger shape than the SLM shape, then the data is
+               cropped to size in a centered manner
+               (:attr:`~slmsuite.holography.toolbox.unpad`).
+             - If integer data is passed with the same type as :attr:`display`
+               (``np.uint8`` for <=8-bit SLMs, ``np.uint16`` otherwise),
+               then this data is **directly** passed to the
+               SLM, without going through the "phase delay to grayscale" conversion
+               defined in the private method :meth:`_phase2gray`. In this situation,
+               ``phase_correct`` and non-zero ``blaze_vector`` are **ignored**.
+               This is error-checked such that bits with greater significance than the
+               bitdepth of the SLM are zero (e.g. the final 6 bits of 16 bit data for a
+               10-bit SLM). Integer data with type different from :attr:`display` leads
+               to an AssertionError.
+
+            Usually, an **exact** stored copy of the data passed by the user under
+            ``phase`` is stored in the attribute :attr:`phase`.
+            However, in cases where :attr:`phase_scaling` not one, this
+            copy is modified to include how the data was wrapped. If the data was
+            cropped, then the cropped data is stored, etc. If integer data was passed,
```

Comment:
I feel like the correct thing to do is to calculate phase information (float) for the passed integer data.

Reply:
Fixed in cdb1a12.